titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/direct/DefaultKubeApiFacade.java

```diff
@@ -0,0 +1,243 @@
+                return;
+            }
+
+            try {
+                this.sharedInformerFactory = createSharedInformerFactory(
+                        "kube-api-server-integrator-shared-informer-",
+                        apiClient
+                );
+
+                this.nodeInformer = createNodeInformer(sharedInformerFactory);
+                this.podInformer = createPodInformer(sharedInformerFactory);
+                this.opportunisticResourceInformer = createOpportunisticResourceInformer(sharedInformerFactory);
+
+                sharedInformerFactory.startAllRegisteredInformers();
+
+                logger.info("Kube node and pod informers activated");
+            } catch (Exception e) {
+                logger.error("Could not initialize Kube client shared informer", e);
+                if (sharedInformerFactory != null) {
+                    ExceptionExt.silent(() -> sharedInformerFactory.stopAllRegisteredInformers());
+                }
+                sharedInformerFactory = null;
+                nodeInformer = null;
+                podInformer = null;
+                throw e;
+            }
+        }
+    }
+
+    private SharedIndexInformer<V1Node> createNodeInformer(SharedInformerFactory sharedInformerFactory) {
+        return sharedInformerFactory.sharedIndexInformerFor(
+                (CallGeneratorParams params) -> coreV1Api.listNodeCall(
+                        null,
+                        null,
+                        null,
+                        null,
+                        null,
+                        params.resourceVersion,
+                        params.timeoutSeconds,
+                        params.watch,
+                        null,
+                        null),
+                V1Node.class,
+                V1NodeList.class,
+                configuration.getKubeApiServerIntegratorRefreshIntervalMs()
+        );
+    }
+
+    private SharedIndexInformer<V1Pod> createPodInformer(SharedInformerFactory sharedInformerFactory) {
+        return sharedInformerFactory.sharedIndexInformerFor(
+                (CallGeneratorParams params) -> coreV1Api.listNamespacedPodCall(
+                        KUBERNETES_NAMESPACE,
+                        null,
+                        null,
+                        null,
+                        null,
+                        null,
+                        params.resourceVersion,
+                        params.timeoutSeconds,
+                        params.watch,
+                        null,
+                        null
+                ),
+                V1Pod.class,
+                V1PodList.class,
+                configuration.getKubeApiServerIntegratorRefreshIntervalMs()
+        );
+    }
+
+    private SharedIndexInformer<V1OpportunisticResource> createOpportunisticResourceInformer(SharedInformerFactory sharedInformerFactory) {
+        // TODO(fabio): enhance the kube client to support custom JSON deserialization options
```

Comment:
you can drop this TODO, I wanted to use jackson instead of gson to get support for immutable POJOs, but it's not worth the effort